### ISSUE TITLE
`railway up` command issues

### ISSUE THREAD

**Issue by GrantBirki** [railwayapp/cli#324]
# `railway up` command issues

For quite sometime, I have been using the following commands in CI (GitHub Actions) without issues. In the last few days, this has started failing. I replicated this issue locally as well and verified that my token is not expired.

```bash
$ export RAILWAY_TOKEN=<token>

$ railway up
Indexing...
Uploading...
  Build Logs: https://railway.app/project/<uuid>/service/<uuid>&
Error: Unauthorized. Please login with `railway login`

$ echo $?
1
```

This command does indeed trigger a new deployment in railway and the deployment completes. However, this command exits with a status of `1` so it always fails CI.

**Comment by Milo123459** [comment]
It should be:
export RAILWAY_TOKEN=<token>

**Comment by GrantBirki** [comment]
@Milo123459 I apologize, that was just a typo in this issue and not what I was actually running locally. I am indeed properly exporting the environment variable.

You can actually see the issue in Actions here as well in our public repo -> https://github.com/the-hideout/stash/actions/runs/4336554752/jobs/7571923972#step:8:8

**Comment by GrantBirki** [comment]
<img width="1189" alt="Screenshot 2023-03-05 at 8 51 40 PM" src="https://user-images.githubusercontent.com/23362539/222985261-e6bf84a7-760d-4091-b768-4074e4f66ff7.png">

**Comment by GrantBirki** [comment]
The way I temporarily fixed this was by forcing our CI to use a `v2.x` version of the railway CLI as this issue is consistent in both our local environment and in CI when using the `v3.x` version of the railway CLI

**Comment by GrantBirki** [comment]
Here is where I updated our code to force a `v2.x` version of the railway/cli

https://github.com/the-hideout/stash/blob/3db9078f32548cddc48254ebc1cc4d445ffc1cde/package.json#L42-L44

**Comment by Sophie-Xinyu-Zhang** [comment]
I'm also running into the same issue. One way I found to go about it might be to log in on the CI manually (need to monitor the actions log), but I think it kills the point of having a CI in the first place. Link: https://github.com/MadeByThePinsHub/setup-railway-cli-action/blob/main/README.md?plain=1#L52-L56

**Comment by Milo123459** [comment]
cc @nebulatgs - do we support RAILWAY_TOKEN?

**Comment by GrantBirki** [comment]
The [railway cli documentation](https://docs.railway.app/develop/cli) does seem to support the usage of `RAILWAY_TOKEN`

<img width="721" alt="Screenshot 2023-03-06 at 9 47 46 AM" src="https://user-images.githubusercontent.com/23362539/223075191-25b0cfd8-66ff-42a1-83f7-30f3cf244fdb.png">

**Comment by Sophie-Xinyu-Zhang** [comment]
Thank you for the quick response, `RAILWAY_TOKEN=XXXX railway run railway up` was able to deploy the server for me. I'm developing a python backend server if that helps anyone

**Comment by Sophie-Xinyu-Zhang** [comment]
I see that even after running the `RAILWAY_TOKEN=*** railway run railway up` command for me, it will still give the same error in the actions log. However, the actions workflow no longer fails and the server was able to deploy. Is this expected?
Here's the actions log of my project:
```
Run RAILWAY_TOKEN=*** railway run railway up
  
  Build Logs: https://railway.app/project/xxx
Error: Unauthorized. Please login with `railway login`
```

**Comment by GrantBirki** [comment]
Something is definitely funky

**Comment by ndneighbor** [comment]
Hey @GrantBirki, @Sophie-Xinyu-Zhang - we are investigating this issue. Apparently it seems to be with how we are operating in a non-interactive environment where the params are outright being ignored. 

cc @nebulatgs

**Comment by nebulatgs** [comment]
Fixed in #387
